### ISSUE TITLE
[Addon Info] - Error when using oneOf propType and a evaluation

### ISSUE THREAD

**Issue by abrahamgnz** [storybooks/storybook#2652]
### Issue details

What did you expect to happen? 
Tried to display the propTable of a component that has a code evaluation in the propTypes declaration.

What happened instead?
Error `propType.value.map is not a function` was displayed

### Steps to reproduce
Create a component with a evaluation in the propTypes declaration.
Add a story withInfo with an object defining the propTables to display the component.
`withInfo({ propTables: [Button] })(() => (...))`
Run the storybook

### Please specify which version of Storybook and optionally any affected addons that you're running

-  `@storybook/react  3.2.12`
-  `@storybook/addon-info  3.2.12`

### Screenshots / Screencast / Code Snippets (Optional)

```
import {values} from 'lodash'

const KIND = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary'
}

const Button = props => (
 ...
)

Button.propTypes = {
/** Type of button **/
  type: oneOf(() => (['hello', 'again'])()),  // <- code evaluation
/** kind of button **/
  kind: oneOf(values(KIND)),  // <- code evaluation
}

```

**Comment by thompsongl** [comment]
Certainly does not solve the underlying problem, but here's a workaround that renders and allows for correct proptype validation:

```js
const KIND = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary'
}
const kindVals = values(KIND)

Button.propTypes = {
  kind: oneOf([...kindVals])
}
```

**Comment by lcsrinaldi** [comment]
Any updates on this issue?

**Comment by Hypnosphi** [comment]
Released [3.3.14](https://github.com/storybooks/storybook/releases/tag/v3.3.14)
